cmd/vpd/src/lib.rs

```diff
@@ -304,21 +205,19 @@ fn vpd_write(
         .context("is the 'vpd' task present?")?;
     let target = target(hubris, subargs)?;
 
-    let mut bytes = vec![];
-
-    if let Some(ref filename) = subargs.write {
+    let bytes = if let Some(ref filename) = subargs.write {
         let file = fs::File::open(filename)?;
 
         let p = tlvc_text::load(file).with_context(|| {
             format!("failed to parse {} as VPD input", filename)
         })?;
 
-        for piece in p {
-            bytes.extend(pack(&piece));
-        }
+        tlvc_text::pack(&p)
     } else {
+        let mut bytes = vec![];
         bytes.resize_with(1024, || 0xffu8);
-    }
+        bytes
```

Comment:
Perhaps replace the whole `else` block with
```rust
    } else {
        vec![0xffu8; 1024]
    }
```

Reply:
Duh -- thanks!